File: algorithms/warmup/python3/diagonal_difference.py
#!/bin/python3

import sys

n = int(input().strip())
a = []
for a_i in range(n):
    a_t = [int(a_temp) for a_temp in input().strip().split(' ')]
    a.append(a_t)

primary_diagonal_sum = 0
secondary_diagonal_sum = 0
for i in range(n):
    primary_diagonal_sum += a[i][i]
    secondary_diagonal_sum += a[i][n - i - 1]
print(abs(primary_diagonal_sum - secondary_diagonal_sum))

File: algorithms/warmup/C#/DiagonalDifference.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

class Solution {

    static void Main(String[] args) {
        int n = Convert.ToInt32(Console.ReadLine());
        int[][] a = new int[n][];
        for(int a_i = 0; a_i < n; a_i++){
           string[] a_temp = Console.ReadLine().Split(' ');
           a[a_i] = Array.ConvertAll(a_temp,Int32.Parse);
        }    
            
        var primaryDiagonalSum = 0;
        var secondaryDiagonalSum = 0;
        for (int i = 0; i < n; i++) 
        {
            primaryDiagonalSum += a[i][i];
            secondaryDiagonalSum += a[i][n - i - 1];
        }
        Console.WriteLine(Math.Abs(primaryDiagonalSum - secondaryDiagonalSum));
    }
}
